pallets/dactr/src/lib.rs

```diff
@@ -130,7 +132,21 @@ pub mod pallet {
 		) -> DispatchResultWithPostInfo {
 			let who = ensure_signed(origin)?;
 			ensure!(!data.is_empty(), Error::<T>::DataCannotBeEmpty);
-			Self::deposit_event(Event::DataSubmitted { who, data });
+
+			// SAFETY: `data.len()` is always less than `u32::MAX` because it is bounded by
+			// `BoundedVec`
+			let len = data.len() as u32;
+
+			// Index Tx in DB block.
+			let data_hash = blake2_256(&data);
+			let extrinsic_index = <frame_system::Pallet<T>>::extrinsic_index()
+				.ok_or_else(|| Error::<T>::BadContext)?;
+			transaction_index::index(extrinsic_index, len, data_hash);
```

Comment:
For my understanding, where is this index used? Is this unique? Can I use this to query my txn?

Reply:
- The index is used internally in the DB. The bug is that we was merging Indexed (regular Tx) & Non Indexed (our submit_data tx) in the same block.
- Yes, it is unique. Same code is used in other non-state-change extrinsics like `Remark/store` transaction in `substrate/frame/remak/lib.rs`.
- It is used internally, but I did not see any public API/SDK yet (at least in v0.9.37). Most likelihood we'll see it in the feature.
att. @prabal-banerjee